vlib/v/checker/tests/match_generic_case_err.out

```diff
@@ -0,0 +1,7 @@
+vlib/v/checker/tests/match_generic_case_err.vv:16:4: error: return type mismatch, it should be `int`
```

Comment:
That seems a bit unclear 🤔 
Can the error also show the type that the compiler thinks `v` is too?
Something like this can be less puzzling:
```
vlib/v/checker/tests/match_generic_case_err.vv:16:4: error: return type mismatch, it should be `int`, but it is instead `string`
```

Reply:
(not in the same PR, but in general)

Reply:
especially for generic code, I think it will help a lot

Reply:
I do agree.